plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-product-related-products-controller.php

```diff
@@ -0,0 +1,254 @@
+<?php
+/**
+ * REST API related-products controller
+ *
+ * Handles requests to the /products/<id>/related-products endpoints.
+ *
+ * @package WooCommerce\RestApi
+ * @since   3.0.0
+ */
+
+use Automattic\WooCommerce\Utilities\I18nUtil;
+
+defined( 'ABSPATH' ) || exit;
+
+use Automattic\Jetpack\Constants;
+
+/**
+ * REST API related-products controller class.
+ *
+ * @package WooCommerce\RestApi
+ * @extends WC_REST_Product_Related_Products_V2_Controller
+ */
+class WC_REST_Product_Related_Products_Controller extends WC_REST_Products_V2_Controller {
```

Comment:
I am a bit torn about creating a separate controller for this mostly because all the controllers are distinctly for other data types, this controller seems to be the only exception to that rule.
Would you be opposed to actually moving this to the `WC_REST_Products_Controller` ? and just registering another route.